### ISSUE TITLE
Openstack Private SSH Timeouts

### ISSUE THREAD

**Issue by jt-johnson** [mitchellh/packer#1855]
- Ubuntu 14.04
- Packer 0.7.5

I'm having problems getting Ubuntu installed on Openstack.  Log snippets and json file are below.

I looked at the boot_command options but it's not available for Openstack so I can't push the install keystrokes needed to get Ubuntu to a point of allowing SSH login's.  The IP shown below is being assigned to the new instance via Openstack - that part is working but I just can't seem to get an OS built past any point.  

Any thoughts?  Let me know if I can provide any more information to help out.  

```text
2015/01/21 14:24:20 ui: openstack output will be in this color.
2015/01/21 14:24:20 ui: 
2015/01/21 14:24:20 Build debug mode: false
2015/01/21 14:24:20 Force build: false
2015/01/21 14:24:20 Preparing build: openstack
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Plugin maximum port: 25000
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Plugin address: unix /tmp/packer-plugin887293152
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Waiting for connection...
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Serving a plugin connection...
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Config: {PackerConfig:{PackerBuildName:openstack PackerBuilderType:openstack PackerDebug:false PackerForce:false PackerUserVars:map[]} AccessConfig:{Username:xxxxxxx Password:xxxxxxx ApiKey: Project: Provider:http://10.10.234.10:5000/v2.0 RawRegion: ProxyUrl: TenantId: Insecure:false} ImageConfig:{ImageName:JT.Test.Image1} RunConfig:{SourceImage:ba21f575-cc96-4278-ab96-6d790d347761 Flavor:d63bf2a0-68b3-4a7d-b2dc-457eb521341f RawSSHTimeout:5m SSHUsername:root SSHPort:22 OpenstackProvider: UseFloatingIp:false FloatingIpPool: FloatingIp: SecurityGroups:[] Networks:[79e653d6-ed19-495c-ba2a-1ed8c06102e2] sshTimeout:300000000000} tpl:0xc2080a9d40}
2015/01/21 14:24:20 Starting build run: openstack
2015/01/21 14:24:20 Running builder: openstack
2015/01/21 14:24:20 Waiting on builds to complete...
2015/01/21 14:24:20 ui: ==> openstack: Creating temporary keypair for this instance...
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 temporary keypair name: packer 54c00af4-4c6a-d430-ea04-86ffd95c1bb0
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 server id: 104bb27b-d5a5-49e5-952a-4de9e2ffb983
2015/01/21 14:24:20 ui: ==> openstack: Waiting for server (104bb27b-d5a5-49e5-952a-4de9e2ffb983) to become ready...
2015/01/21 14:24:20 packer-builder-openstack: 2015/01/21 14:24:20 Waiting for state to become: ACTIVE
2015/01/21 14:24:21 packer-builder-openstack: 2015/01/21 14:24:21 Waiting for state to become: ACTIVE currently BUILD (0%)
2015/01/21 14:24:23 packer-builder-openstack: 2015/01/21 14:24:23 Waiting for state to become: ACTIVE currently BUILD (0%)
2015/01/21 14:24:25 packer-builder-openstack: 2015/01/21 14:24:25 Waiting for state to become: ACTIVE currently BUILD (0%)
2015/01/21 14:24:27 packer-builder-openstack: 2015/01/21 14:24:27 Waiting for SSH, up to timeout: 5m0s
2015/01/21 14:24:27 ui: ==> openstack: Waiting for SSH to become available...
2015/01/21 14:24:43 packer-builder-openstack: 2015/01/21 14:24:43 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:25:03 packer-builder-openstack: 2015/01/21 14:25:03 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:25:24 packer-builder-openstack: 2015/01/21 14:25:24 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:25:44 packer-builder-openstack: 2015/01/21 14:25:44 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:26:05 packer-builder-openstack: 2015/01/21 14:26:05 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:26:25 packer-builder-openstack: 2015/01/21 14:26:25 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:26:46 packer-builder-openstack: 2015/01/21 14:26:46 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:27:07 packer-builder-openstack: 2015/01/21 14:27:07 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:27:27 packer-builder-openstack: 2015/01/21 14:27:27 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:27:48 packer-builder-openstack: 2015/01/21 14:27:48 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:28:08 packer-builder-openstack: 2015/01/21 14:28:08 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:28:29 packer-builder-openstack: 2015/01/21 14:28:29 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:28:49 packer-builder-openstack: 2015/01/21 14:28:49 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:29:10 packer-builder-openstack: 2015/01/21 14:29:10 TCP connection to SSH ip/port failed: dial tcp 10.10.235.129:22: i/o timeout
2015/01/21 14:29:27 ui error: ==> openstack: Timeout waiting for SSH.
2015/01/21 14:29:27 ui: ==> openstack: Terminating the source server...
2015/01/21 14:29:27 packer-builder-openstack: 2015/01/21 14:29:27 Waiting for state to become: DELETED
2015/01/21 14:29:27 packer-builder-openstack: 2015/01/21 14:29:27 Waiting for state to become: DELETED currently ACTIVE (0%)
2015/01/21 14:29:29 packer-builder-openstack: 2015/01/21 14:29:29 404 on ServerStateRefresh, returning DELETED
2015/01/21 14:29:29 ui: ==> openstack: Deleting temporary keypair...
2015/01/21 14:29:30 ui error: Build 'openstack' errored: Timeout waiting for SSH.
2015/01/21 14:29:30 Builds completed. Waiting on interrupt barrier...
2015/01/21 14:29:30 machine readable: error-count []string{"1"}
2015/01/21 14:29:30 ui error: 
==> Some builds didn't complete successfully and had errors:
2015/01/21 14:29:30 machine readable: openstack,error []string{"Timeout waiting for SSH."}
2015/01/21 14:29:30 ui error: --> openstack: Timeout waiting for SSH.
2015/01/21 14:29:30 ui: 
==> Builds finished but no artifacts were created.
2015/01/21 14:29:30 waiting for all plugin processes to complete...
2015/01/21 14:29:30 /packer/packer-builder-openstack: plugin process exited
```

```json
{
   "builders": [
    {
      "username": "xxxxxxx",
      "password": "xxxxxx",
      "provider": "http://10.10.234.10:5000/v2.0",
      "networks": "79e653d6-ed19-495c-ba2a-1ed8c06102e2",
      "flavor": "d63bf2a0-68b3-4a7d-b2dc-457eb521341f",
      "image_name": "Test.Image1",
      "source_image": "ba21f575-cc96-4278-ab96-6d790d347761",
      "type": "openstack"
    
    }]
  }
```

**Comment by jenspinney** [comment]
Have you tried using a floating IP? (Set "use_floating_ip" to true and give the name of your floating IP pool in "floating_ip_pool").

**Comment by sethvargo** [comment]
Hi @jt-johnson - did you see @jenspinney's comment?

**Comment by sci-burns** [comment]
Had the same issue, allocated a floating IP from the pool and it worked like a champ.  Of course the next problem is that the floating IP address from the pool is not released - if a new address is allocated it should be cleaned up.

**Comment by mitchellh** [comment]
Thanks @jenspinney. We're also going to add private IP support if you're connecting via VPN, but another issue is tracking that.

**Comment by vallard** [comment]
is private IP support already added?  I have set up my cluster with 'provider' networking, so there are no floating IP addresses, just a network I'm connected to.  When it gets to "Waiting for SSH to become available" I'm able to ssh into it from another terminal but packer continues to wait until it times out.  Thx.

**Comment by vallard** [comment]
Figured this out: https://github.com/mitchellh/packer/issues/2526
